### PR TITLE
JDK-8307349: Support xlc17 clang toolchain on AIX

### PR DESCRIPTION
**Pull request by JoKern65**
The new xlc17 compiler should be supported to build OpenJDK on AIX. This compiler, compared to the currently supported xlc16, has a significantly more recent clang (xlc 17.1.1 uses clang 15) included.
1. Because the frontend interface of the new compiler (c-flags, Ld-Flags) has changed from an xlc to a clang interface we decided to use the clang toolchain for the new xlc17 compiler.
2. Unfortunately, the system headers are mainly unchanged, so they do not harmonize with the src/hotspot/share/utilities/globalDefinitions_gcc.hpp which would be used if we totally switch to clang toolchain. So we keep the HOTSPOT_TOOLCHAIN_TYPE=xlc
3. In src/hotspot/share/utilities/globalDefinitions_xlc.hpp we introduce a new define AIX_XLC_GE_17 which is set if we build with the new xlc17 on AIX. This define will be used in following PRs.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307349](https://bugs.openjdk.org/browse/JDK-8307349): Support xlc17 clang toolchain on AIX

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [a94aa615](https://git.openjdk.org/jdk/pull/13898/files/a94aa615b11390a503b65771d84f4f14c69c7552)
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13898/head:pull/13898` \
`$ git checkout pull/13898`

Update a local copy of the PR: \
`$ git checkout pull/13898` \
`$ git pull https://git.openjdk.org/jdk.git pull/13898/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13898`

View PR using the GUI difftool: \
`$ git pr show -t 13898`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13898.diff">https://git.openjdk.org/jdk/pull/13898.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13898#issuecomment-1541964586)

### CODE REVIEW

**Comment by MBaesken** [comment]
The copyright year info needs updating to 2023 in some files (please look at the start of the files).
e.g. make/autoconf/flags-ldflags.m4

File: make/hotspot/lib/JvmOverrideFiles.gmk [hunk 1]
```diff
@@ -109,7 +109,11 @@ else ifeq ($(call isTargetOs, macosx), true)
   endif
 
 else ifeq ($(call isTargetOs, aix), true)
-  BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -qnoinline
+  ifeq ($(TOOLCHAIN_TYPE), clang)
+   BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -fno-inline
+  else
+   BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -qnoinline
+  endif
```
Comment (erikj79) [inline make/hotspot/lib/JvmOverrideFiles.gmk:116]:
```suggestion
  ifeq ($(TOOLCHAIN_TYPE), clang)
    BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -fno-inline
  else
    BUILD_LIBJVM_synchronizer.cpp_CXXFLAGS := -qnoinline
  endif
```

**Review by erikj79 — APPROVED** [review]
Looks ok, just some whitespace suggestions.

File: src/hotspot/share/utilities/globalDefinitions_xlc.hpp
Comment (MBaesken) [inline src/hotspot/share/utilities/globalDefinitions_xlc.hpp:1]:
```
  #if __open_xl_version__ < 17
  #error "xlc < 16 not supported"
  #endif
```
Should this be xlc < 17 in the error ?

**Comment by JoKern65** [comment]
I followed your suggested corrections. Thanks a lot.

**Pull request closed without merging**